UT4MasterServer.Services/UT4MasterServer.Services.csproj

```diff
@@ -7,8 +7,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="7.0.0" />
-    <PackageReference Include="Microsoft.Extensions.Options" Version="7.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" Version="8.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Options" Version="8.0.2" />
```

Comment:
Here too. Please follow package named from docs: https://learn.microsoft.com/en-us/aspnet/core/migration/70-80?view=aspnetcore-8.0&tabs=visual-studio#update-package-references

Reply:
This is confusing.

`Microsoft.Extensions.*` packages on latest versions (*8.\*.\**) have `net6.0` listed in dependencies.
`Microsoft.AspNetCore.*` packages on latest versions (*8.\*.\**) have only `net8.0`.

Is it preferred to use *8.\*.\** versions even when targeting .NET 6?
Were you in previous comment expecting me to change all packages to absolute latest versions or to *6.\*.\**?

Reply:
Even though `Microsoft.AspNetCore.*` has only `net8.0` as dependency NuGet Manager allows to update to it. Should I just let manager update whatever it can to latest version?

Reply:
>Even though Microsoft.AspNetCore.* has only net8.0 as dependency NuGet Manager allows to update to it.

It will allow, but installation should fail with error about wrong targetframework

I'm still looking for one answer from Microsoft about these packages. IIRC he said that updating packages 

But you are right -> Extensions are more independent. For example: https://www.nuget.org/packages/Microsoft.Extensions.Options/#dependencies-body-tab

Reply:
https://www.reddit.com/r/csharp/comments/l6se45/should_i_update_my_microsoft_packages_to_version/

Reply:
Found it: https://github.com/dotnet/extensions/issues/2700#issuecomment-580413093

Reply:
> It will allow, but installation should fail with error about wrong targetframework

You say that it should fail, but it didn't for me. I was able to successfully build and run whole solution.

> Found it: [dotnet/extensions#2700 (comment)](https://github.com/dotnet/extensions/issues/2700#issuecomment-580413093)

The page you linked answers only how to handle `Microsoft.Extensions.*` packages. It says they need to have the same major version, and we do. All `Microsoft.Extensions.*` have major version 8.

Only author of the question mentioned that you should generally follow semantic versioning based on targetframework, but source of this information is unknown and therefore this information is not trustworthy.

I am just wondering if package major versions are in any way related to targetframework? I'm starting to think that they are not because I am able to build when most packages are on latest version that nuget allows to update to even though all our projects target `net6.0`.

But just in case, yes, lets stick to major version 6 on all `Microsoft.*` packages.

Reply:
This change in `Microsoft.Extensions.*` packages will essentially be a downgrade because we were using major version 7 before.